crates/filter/src/source.rs

```diff
@@ -7,34 +8,40 @@ use subprocess::Exec;
 
 /// Source is anything that can produce an iterator of String.
 #[derive(Debug)]
-pub enum Source<I: Iterator<Item = String>> {
+pub enum Source<I: Iterator<Item = SourceItem>> {
     Stdin,
     #[cfg(feature = "enable_dyn")]
     Exec(Exec),
     File(PathBuf),
     List(I),
 }
 
-impl From<Vec<String>> for Source<std::vec::IntoIter<String>> {
+impl From<Vec<String>> for Source<std::vec::IntoIter<SourceItem>> {
     fn from(source_list: Vec<String>) -> Self {
-        Self::List(source_list.into_iter())
+        // XXX: Not sure about the .into_iter().map().collect().into_iter(), pretty sure it's doing
+        // something wasteful.
+        Self::List(source_list
+            .into_iter()
+            .map(|line| SourceItem { display: line, filter: None })
+            .collect::<Vec<_>>()
+            .into_iter())
```

Comment:
I don't like this impl at all. Probably `Source` should have a function like this:

```rust
fn string_into_source_item(s: String) -> SourceItem;

pub fn from_iter<I>(iter: I) -> Source<std::iter::Map<I::IntoIter, string_into_source_item>>
where I: IntoIterator<Item = String>
{
    Self::List(iter.into_iter().map(string_into_source_item))
}
```

Reply:
Still not the best thing to do, but I'm too lazy to fix it properly.